### PR TITLE
Enable website publishing in CI

### PR DESCRIPTION
**Pull request by danicheg**

### CODE REVIEW

File: .github/workflows/publish-website.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: coursier/cache-action@v6
+      - name: Publish ${{ github.ref }}
+        run: sbt docs/docusaurusPublishGhpages
+        env:
+          GITHUB_DEPLOY_KEY: ${{ secrets.GITHUB_DEPLOY_KEY }}
```
Comment (danicheg) [inline .github/workflows/publish-website.yml:18]: I lack access to this part of the repo's settings in Github, so I need some assistance setting this token up https://scalameta.org/mdoc/docs/docusaurus.html#deploy-key Maybe @SethTisue can do that?

Comment (SethTisue) [reply .github/workflows/publish-website.yml:18]: I followed the steps, accurately I hope :-)

**Comment by SethTisue** [comment]
https://github.com/scalacenter/scalac-profiling/actions/runs/7441157666/job/20242939961 was automatically triggered when I merged this, but it seems to have failed for reasons I don't understand. @danicheg any insight?

**Comment by danicheg** [comment]
@SethTisue Thank you!
In the logs, I see this
```
Cloning into 'scalac-profiling-gh-pages'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.
```
Have you enabled the `Allow write access` in the settings as specified in [docs](https://scalameta.org/mdoc/docs/docusaurus.html#deploy-key)?

**Comment by SethTisue** [comment]
I did use an empty passphrase.

You mean `GIT_DEPLOY_KEY`, not `GITHUB_DEPLOY_KEY`. It appears correct to me at https://github.com/scalacenter/scalac-profiling/settings/secrets/actions

> Have you enabled the Allow write access in the settings as specified in [docs](https://scalameta.org/mdoc/docs/docusaurus.html#deploy-key)?

Yes, at https://github.com/scalacenter/scalac-profiling/settings/keys I see it says "Read/write"

Maybe the clone is failing because the gh-pages branch doesn't exist yet? Maybe you need to create it first?

**Comment by danicheg** [comment]
> Maybe the clone is failing because the gh-pages branch doesn't exist yet?
Screenshot 2024-01-08 at 19 12 54

**Comment by danicheg** [comment]
> You mean GIT_DEPLOY_KEY, not GITHUB_DEPLOY_KEY. It appears correct to me at https://github.com/scalacenter/scalac-profiling/settings/secrets/actions

I cannot visit that page. But just to confirm, you've set up the `GIT_DEPLOY_KEY` env var, right? If so it might be the reason, because in CI, I expected this one `${{ secrets.GITHUB_DEPLOY_KEY }}`

**Comment by SethTisue** [comment]
Yes, that must be the problem. I used `GIT_DEPLOY_KEY` because that's what https://scalameta.org/mdoc/docs/docusaurus.html#deploy-key suggested. 

Hopefully https://github.com/scalacenter/scalac-profiling/pull/77 will fix it

**Pull request merged**